node/node.go

```diff
@@ -85,13 +84,6 @@ type Node struct {
 // New sets the config and logger to the StarknetNode.
 // Any errors while parsing the config on creating logger will be returned.
 func New(cfg *Config, version string) (*Node, error) { //nolint:gocyclo,funlen
-	if cfg.DatabasePath == "" {
-		dirPrefix, err := utils.DefaultDataDir()
-		if err != nil {
-			return nil, err
-		}
-		cfg.DatabasePath = filepath.Join(dirPrefix, cfg.Network.String())
-	}
```

Comment:
Why not set the default db path here instead? if `pwd, err := os.Getwd()` fails then we can return the error.

Reply:
Then we print the incorrect default directory on `--help` and in the DEBUG log message. (also, it would be very nice if all config values are set before we enter `node.New`, rather than changing the defaults on the fly)

Reply:
We can call `os.Getwd()` in the `PreExec` function and return the error. That way `--help` succeeds, but running Juno will fail. I'll implement that.

Reply:
Done.